crates/nu-parser/src/lite_parse.rs

```diff
@@ -55,6 +55,12 @@ enum BlockKind {
     SquareBracket,
 }
 
+#[cfg(target_family = "windows")]
+const ESCAPE_CHAR: char = '^';
```

Comment:
@jonathandturner brought up another good point in Discord about why we've been not wanting to do a backslash escape: it conflicts with Windows paths. You've taken care of that here, but there's still a few problems:

1. A caret doesn't look super nice, but we could also just be so used to seeing backslashes in the middle of bare words that we consider them ok to look at haha.
2. `^` is used to force running an external command instead of an internal one (e.g., `^echo ...`). In the middle of a bare word this wouldn't be confusing, but I worry there's too much potential for ambiguity.
3. It creates an inconsistent experience across platforms. I don't think we've called it out as a tenet anywhere, but nushells strives to have a consistent experience across platforms, as much as is possible. If we use `^` as an escape character on Windows, it should be the escape character everywhere.

Reply:
How do you want to handle this tho? You need something as an escape char.

I'd suggest switching windows over to use `/` as separator and unifying the experience...

Reply:
I guess one option would be to not do escaping at all, and enforce quoting, including from the completion values.

Reply:
> You need something as an escape char.

It definitely ads conciseness, but it's not needed. This is verbose, but shows that escape characters aren't necessary:

```
echo `test{{$(char single_quote)}}`
```

You can also quote things with a different quote character than what you'd want to escape:

```
echo `I can use "double quotes" and 'single quotes' in here`
```

Now, I can understand not wanting to type that out, but we have three quoting characters, and the likelihood of needing to construct a string that contains all likely very low. With tab completion that works, you won't even have to think about it how to go about quoting either.

The reason I'm giving you pushback here, is the same reason we've given ourselves pushback: we have enough momentum and users now that [Hyrum's Law](https://www.hyrumslaw.com/) kicks in. The longer we can defer decisions, the better we can work out the best solution. Escaping is one of several things we are trying _really_ hard to avoid, because we want to find a nicer, more expressive way to write these special characters.

> The problem with that is incompatibility with many existing scripts out there, where people are used to using \ so people will complain things don't work in nushell when they C&P some script over.

nushell will not be compatible with any existing scripts, meaning nushell isn't going to be able to interpret a shell script from existing shells. We're trying really hard to re-envision the shell, which means doing things differently, even if it feels a little awkward at first. If one really wants to use code/aliases/commands from other shells, there are two options:

1. rewrite them to be compatible with nushell, or
2. put them in a file with a `/path/to/my/shell -c` shebang.